schema/snapcraft.yaml

```diff
@@ -253,6 +253,24 @@ properties:
               - forking
               - oneshot
               - notify
+          refresh-mode:
+              type: string
+              description: controls if the app should be restarted at all
+              enum:
+                - endure
+                - restart
+          stop-mode:
```

Comment:
Both `refresh-mode` and `stop-mode` can only be used for daemons. You can probably use dependencies for this to validate in the schema.